File: 2124-check-if-all-as-appears-before-all-bs/2124-check-if-all-as-appears-before-all-bs.c
bool checkString(char * s){
    int i = 0, pos_a = -1, pos_b = -1;
    
    while(s[i]){
        if(s[i] == 'a')  //max position for a
            pos_a = i;
        else if(s[i] == 'b' && pos_b == -1)  //min position for b
            pos_b = i;
        i++;
    }
    return (pos_b > pos_a || pos_b == -1);   
}

File: 2124-check-if-all-as-appears-before-all-bs/2124-check-if-all-as-appears-before-all-bs.swift
class Solution {
    func checkString(_ s: String) -> Bool {
        var flag = false
        
        for char in s {
            if char == "b" { flag = true }
            else {
                if flag {
                    if char == "a" { return false}
                }
            }
        }
        
        return true
    }
}